python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -707,6 +712,15 @@ def asset_key_for_input(self, node_handle: NodeHandle, input_name: str) -> Optio
     def io_manager_key_for_asset(self, asset_key: AssetKey) -> str:
         return self._io_manager_keys_by_asset_key.get(asset_key, "io_manager")
 
+    def is_source_for_asset(self, asset_key: AssetKey) -> bool:
+        return asset_key in self.source_assets_by_key
+
+    def is_versioned_for_asset(self, asset_key: AssetKey) -> bool:
+        return self._assets_defs_by_key[asset_key].is_versioned
+
+    def op_version_for_asset(self, asset_key: AssetKey) -> Optional[str]:
```

Comment:
Seems like this needs to be generalized to `node_version_for_asset` for graph-backed assets